public/class-perfecty-push-users.php

```diff
@@ -77,8 +77,9 @@ public function register( $data ) {
 				$user = Perfecty_Push_Lib_Db::get_user( $result );
 
 				// Send a confirmation notification.
-				if ( $first_time ) {
-					$payload = Perfecty_Push_Lib_Payload::build( "Congratulations, you're now subscribed!" );
+				$options = get_option( 'perfecty_push' );
+				if ( $first_time && ( 1 === $options['settings_send_welcome_message'] ) ) {
```

Comment:
You need to check if the option exists like:

```php
$send_welcome_message = isset( $options['segmentation_enabled'] ) && $options['segmentation_enabled'] == 1;
if ( $first_time && $send_welcome_message ) {
   ...
```

Reply:
Also, $payload is built with `$payload = Perfecty_Push_Lib_Payload::build( "Congratulations, you're now subscribed!" );`